src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,106 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) "
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                         = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getObject(1, Long.class));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT id, title, price FROM books WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Book book = null;
+            if (resultSet.next()) {
+                book = getBook(resultSet);
+            }
```

Comment:
Use `if` or `while` with `resultSet.next()`. Don't do it without checking, because the result can be `null` and you will get a NPE after trying to get a value from `resultSet`. In this case, you have implemented the check correctly in `findById` method.